conans/client/graph/grapher.py

```diff
@@ -64,25 +26,67 @@ def html_binary_graph(search_info, table_filename):
 </script>
 <h1>%s</h1>
     """ % str(reference)]
-    headers = sorted(columns)
-    result.append("<table>")
-    result.append("<tr>")
-    result.append("<th></th>")
-    for header in headers:
-        result.append("<th>%s</th>" % header)
-    result.append("</tr>")
-    for row, columns in sorted(table.items()):
+
+    for remote_info in search_info:
+        if remote_info["remote"] != 'None':
+            result.append("<h2>'%s':</h2>" % str(remote_info["remote"]))
+
+        ordered_packages = remote_info["items"][0]["packages"]
+        binary = namedtuple("Binary", "ID outdated")
+        columns = set()
+        table = defaultdict(dict)
+        for package in ordered_packages:
+            package_id = package["id"]
+            settings = package["settings"]
+            if settings:
+                row_name = "%s %s %s" % (settings.get("os", "None"), settings.get("compiler", "None"),
+                                        settings.get("compiler.version", "None"))
+                column_name = []
+                for setting, value in settings.items():
+                    if setting.startswith("compiler."):
+                        if not setting.startswith("compiler.version"):
+                            row_name += " (%s)" % value
+                    elif setting not in ("os", "compiler"):
+                        column_name.append(value)
+                column_name = " ".join(column_name)
+            else:
+                row_name = "NO settings"
+                column_name = ""
+
+            options = package["options"]
+            if options:
+                for k, v in options.items():
+                    column_name += "<br>%s=%s" % (k, v)
+
+            column_name = column_name or "NO options"
+            columns.add(column_name)
+            # Always compare outdated with local recipe, simplification,
+            # if a remote check is needed install recipe first
+            if "outdated" in package:
+                outdated = package["outdated"]
+            else:
+                outdated = None
+            table[row_name][column_name] = binary(package_id, outdated)
+
+        headers = sorted(columns)
+        result.append("<table>")
         result.append("<tr>")
-        result.append("<td>%s</td>" % row)
+        result.append("<th></th>")
```

Comment:
Not sure if I get why this empty header <th>

Reply:
It was there before, see previous line 70. I'm not sure is it really needed.